Always use ESSL translator frontend code in ANGLE

kkinnunen-apple opened the pull request:
#### e130e44dc05dda7556e31d3cf23ba2f3e095249f
<pre>
Always use ESSL translator frontend code in ANGLE
<a href="https://bugs.webkit.org/show_bug.cgi?id=245126">https://bugs.webkit.org/show_bug.cgi?id=245126</a>
rdar://problem/100159739

Reviewed by Adrian Perez de Castro.

These files support input of full GLSL
SymbolTable_autogen.cpp
ImmutableString_autogen.cpp
These are useful only when ANGLE is used as (big) OpenGL
implementation.

These files support input of GLSL ES:
SymbolTable_ESSL_autogen.cpp
ImmutableString_ESSL_autogen.cpp
These should be used when ANGLE is used as WebGL or OpenGL ES
implementation.

Decreases the binary size somewhat.

* Source/ThirdParty/ANGLE/ANGLE.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/CMakeLists.txt:

Canonical link: <a href="https://commits.webkit.org/254687@main">https://commits.webkit.org/254687@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d49ac3f54acbf5359bbc65db238636d518d85e30

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/89833 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/34380 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/20518 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/99149 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/155971 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/32871 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/28311 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/82175 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/93499 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/95480 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/26119 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/76652 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/26046 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80995 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/80840 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/69047 "Found 6 new API test failures: /WebKit2Gtk/TestWebKitSettings:/webkit/WebKitSettings/webkit-settings, /WebKit2Gtk/TestWebKitVersion:/webkit/WebKitVersion/version, /WebKit2Gtk/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/children-changed, /WebKit2Gtk/TestLoaderClient:/webkit/WebKitWebPage/redirect-to-data-uri, /WebKit2Gtk/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/basic-hierarchy, /TestWTF:WTF_WordLock.ManyContendedLongSections (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/30621 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/14923 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/30362 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15864 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3291 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/33819 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/38814 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/32534 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34948 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

aperezdc reviewed (approved):
LGTM, thanks @kkinnunen-apple, you were faster than me preparing this one 😸

kkinnunen-apple committed:
Always use ESSL translator frontend code in ANGLE
https://bugs.webkit.org/show_bug.cgi?id=245126
rdar://problem/100159739

Reviewed by Adrian Perez de Castro.

These files support input of full GLSL
SymbolTable_autogen.cpp
ImmutableString_autogen.cpp
These are useful only when ANGLE is used as (big) OpenGL
implementation.

These files support input of GLSL ES:
SymbolTable_ESSL_autogen.cpp
ImmutableString_ESSL_autogen.cpp
These should be used when ANGLE is used as WebGL or OpenGL ES
implementation.

Decreases the binary size somewhat.

* Source/ThirdParty/ANGLE/ANGLE.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/CMakeLists.txt:

Canonical link: https://commits.webkit.org/254687@main
